src/sds_data_model/_dataframe.py

```diff
@@ -21,6 +27,21 @@
 from sds_data_model.metadata import Metadata
 
 
+@udf(returnType=ArrayType(FloatType()))
+def _bng_to_bounds(grid_reference: str
+) -> Tuple[float, float, float, float]:
+    """_summary_
+
+    Args:
+        grid_reference (str): _description_
+
+    Returns:
+        Tuple[float, float, float, float]: _description_
+    """    
+    wkt = wkt_from_bng(grid_reference)
+    return loads(wkt).bounds
```

Comment:
`mypy` is upset here because `shapley`'s `BaseGeometry.bounds` property doesn't specify a return type.  To get round this you can assign to a variable you have explicitly annotated and then return that.

```suggestion
@udf(returnType=ArrayType(FloatType()))
def _bng_to_bounds(grid_reference: str
) -> Tuple[float, float, float, float]:
    """_summary_

    Args:
        grid_reference (str): _description_

    Returns:
        Tuple[float, float, float, float]: _description_
    """    
    wkt = wkt_from_bng(grid_reference)
    bounds: Tuple[ float, float, float, float] = loads(wkt).bounds
    return bounds
```